### PR TITLE
added documentation for automatic updates (mainnet only)

### LINKED ISSUE #232: Add documentation for automatic updates for desktop wallet
From version 1.2 and onwards of the desktop wallet, will (with users acceptance) be updated automatically. When an update is available the desktop wallet will show a notification, allowing the user to either postpone the update or restart and install the update.

The automatic update process does the following: 
- Downloads update
- Verifies integrity of update (matches sha256 checksum + verifies signature using published public key)
- Restarts application if verification i successful.

### PR DESCRIPTION
**Pull request by soerenbf**
## Purpose

From DTW version 1.2 and onwards, the application features automatic updates. This PR ensures our documentation describes how this works, and what to do (manually updating) if this is not supported. 

## Changes

- Added page describing both automatic and manual update process

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.

Closes #232

### CODE REVIEW

File: source/mainnet/net/desktop-wallet/update-application.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
+.. _update-application:
+
+===========================
+Updating the desktop wallet
```
Comment (LotteKh) [inline source/mainnet/net/desktop-wallet/update-application.rst:4]:
```suggestion
Update the desktop wallet
```

File: source/mainnet/net/desktop-wallet/update-application.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+===========================
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+The desktop wallet application will automatically update itself on **MacOS**, **Windows**, and **Linux** (though only for the AppImage distribution).
```
Comment (LotteKh) [inline source/mainnet/net/desktop-wallet/update-application.rst:12]:
```suggestion
The Desktop Wallet installs updates automatically on **MacOS**, **Windows**, and **Linux** (though only for the AppImage distribution).
```

File: source/mainnet/net/desktop-wallet/update-application.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+Automatic updates
+=================
+
+When notified about an update being available, you can select either:
```
Comment (LotteKh) [inline source/mainnet/net/desktop-wallet/update-application.rst:17]:
```suggestion
The Desktop Wallet notifies you when an update is available, and you then have the following options:
```

File: source/mainnet/net/desktop-wallet/update-application.rst [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+* **Restart and Install** to install the update, or
```
Comment (LotteKh) [inline source/mainnet/net/desktop-wallet/update-application.rst:19]:
```suggestion
* **Restart and Install** to install the update
```

File: source/mainnet/net/desktop-wallet/update-application.rst [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+* **Remind me** to be notified again the next time you open the application.
+
+If you choose to update the application, the update process will automatically ensure the integrity of the downloaded files before applying the update.
```
Comment (LotteKh) [inline source/mainnet/net/desktop-wallet/update-application.rst:22]:
```suggestion
If you choose to update the application, the update process automatically ensures the integrity of the downloaded files before applying the update.
```

File: source/mainnet/net/desktop-wallet/update-application.rst [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+.. note::
+    If the integrity of an update cannot be verified, the update is rejected. The verification step is there to make sure the update is an official release from Concordium, and has not been tampered with by malware.
+
+Manually updating
+=================
+
+To manually update the desktop wallet, simply go to the :ref:`downloads page<downloads-desktop-wallet>`, to download the latest version.
```
Comment (LotteKh) [inline source/mainnet/net/desktop-wallet/update-application.rst:30]:
```suggestion
To manually update the desktop wallet, go to the :ref:`downloads page<downloads-desktop-wallet>` and download the latest version.
```

**Review by LotteKh — CHANGES_REQUESTED** [review]

File: source/mainnet/net/desktop-wallet/update-application.rst (hunk 1 above)
Comment (soerenbf) [reply source/mainnet/net/desktop-wallet/update-application.rst:4]: I don't think I get this. In my opinion "Update ..." sounds like a request to the reader, where as "Updating ..." sounds more like an introduction to a topic?

Comment (LotteKh) [reply source/mainnet/net/desktop-wallet/update-application.rst:4]: It's not standard in documentation anymore to start with an -ing verb or a gerund (it used to be a couple of years ago). It's more common to use the infinitive. Alternatively, you could write Automatic updates of the Desktop Wallet or How to update the Desktop Wallet.

Comment (soerenbf) [reply source/mainnet/net/desktop-wallet/update-application.rst:4]: I'll leave it for you to decide, so I'll apply the suggestion 🙂 thanks for clarifying.

**Review by LotteKh — APPROVED** [review]

**Pull request merged**